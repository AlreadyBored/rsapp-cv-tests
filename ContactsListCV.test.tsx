import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ContactsListCV from '../ContactsListCV';

describe('ContactsListCV', () => {
  describe('Should render correctly', () => {
    it('with no contacts data', () => {
      const output = shallow(<ContactsListCV contacts={null} />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('with all filled contacts', () => {
      const contactsData = {
        phone: '+375111111',
        email: 'test@test.com',
        skype: 'test_skypeid',
        telegram: 'test_tg',
        linkedin: 'test',
        locations: 'Test, Test',
        github: 'test',
        website: 'https://test.com',
      };

      const output = shallow(<ContactsListCV contacts={contactsData} />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('with excess contacts data', () => {
      const contactsData = {
        phone: '+375111111',
        email: 'test@test.com',
        skype: 'test_skypeid',
        telegram: 'test_tg',
        linkedin: 'test',
        locations: 'Test, Test',
        github: 'test',
        website: 'https://test.com',
        someOtherContact: 'test',
        test: 'test',
        foo: 'bar',
        bar: 'baz',
      };

      const output = shallow(<ContactsListCV contacts={contactsData} />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('with 3 locations', () => {
      const contactsData = {
        phone: '+375111111',
        email: 'test@test.com',
        skype: 'test_skypeid',
        telegram: 'test_tg',
        linkedin: 'test',
        locations: 'Test, Test;Test2, Test2;Test3, Test3',
        github: 'test',
        website: 'https://test.com',
      };

      const output = shallow(<ContactsListCV contacts={contactsData} />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });
});
