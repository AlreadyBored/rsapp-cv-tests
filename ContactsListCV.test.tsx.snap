// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsListCV Should render correctly with 3 locations 1`] = `
<List
  dataSource={
    Array [
      Array [
        "phone",
        "+375111111",
      ],
      Array [
        "email",
        "test@test.com",
      ],
      Array [
        "skype",
        "test_skypeid",
      ],
      Array [
        "telegram",
        "test_tg",
      ],
      Array [
        "linkedin",
        "test",
      ],
      Array [
        "locations",
        "Test, Test;Test2, Test2;Test3, Test3",
      ],
      Array [
        "github",
        "test",
      ],
      Array [
        "website",
        "https://test.com",
      ],
    ]
  }
  renderItem={[Function]}
  size="small"
/>
`;

exports[`ContactsListCV Should render correctly with all filled contacts 1`] = `
<List
  dataSource={
    Array [
      Array [
        "phone",
        "+375111111",
      ],
      Array [
        "email",
        "test@test.com",
      ],
      Array [
        "skype",
        "test_skypeid",
      ],
      Array [
        "telegram",
        "test_tg",
      ],
      Array [
        "linkedin",
        "test",
      ],
      Array [
        "locations",
        "Test, Test",
      ],
      Array [
        "github",
        "test",
      ],
      Array [
        "website",
        "https://test.com",
      ],
    ]
  }
  renderItem={[Function]}
  size="small"
/>
`;

exports[`ContactsListCV Should render correctly with excess contacts data 1`] = `
<List
  dataSource={
    Array [
      Array [
        "phone",
        "+375111111",
      ],
      Array [
        "email",
        "test@test.com",
      ],
      Array [
        "skype",
        "test_skypeid",
      ],
      Array [
        "telegram",
        "test_tg",
      ],
      Array [
        "linkedin",
        "test",
      ],
      Array [
        "locations",
        "Test, Test",
      ],
      Array [
        "github",
        "test",
      ],
      Array [
        "website",
        "https://test.com",
      ],
      Array [
        "someOtherContact",
        "test",
      ],
      Array [
        "test",
        "test",
      ],
      Array [
        "foo",
        "bar",
      ],
      Array [
        "bar",
        "baz",
      ],
    ]
  }
  renderItem={[Function]}
  size="small"
/>
`;

exports[`ContactsListCV Should render correctly with no contacts data 1`] = `""`;
